src/test/java/seedu/mindmymoney/command/UpdateCommandTest.java

```diff
@@ -44,4 +52,120 @@ void updateCommand_invalidInput_exceptionThrown() {
         UpdateCommand updateCommand = new UpdateCommand(input, testUser);
         assertThrows(MindMyMoneyException.class, updateCommand::executeCommand);
     }
+
+    /**
+     * Assert that when the update command fields are similar to the expenditure in the list, an exception is thrown.
+     *
+     * */
+    @Test
+    void updateExpenditureCommand_updateFieldSimilarToExpenditureInList_exceptionThrown() {
+        Expenditure testExpenditure = new Expenditure("Cash", "Food",
+                "porridge", 5, "01/04/2022");
+        User testUser = new User();
+        testUser.setExpenditureListArray(new ExpenditureList());
+        testUser.getExpenditureListArray().add(testExpenditure);
+        String input = "1 /pm cash /c food /d porridge /a 5 /t 01/04/2022";
+        UpdateCommand updateCommand = new UpdateCommand(input,testUser);
+        assertThrows(MindMyMoneyException.class, updateCommand::executeCommand);
+    }
+
+    /**
+     * Assert that the command can update the credit card list.
+     * */
+    @Test
+    void updateCreditCardCommand_updateCreditCard_listUpdated() {
+        CreditCard testCreditCard = new CreditCard("DBS", 2, 1000, 1000);
+        CreditCard newCreditCard = new CreditCard("DBS", 5, 2000, 2000);
+        User testUser = new User();
+        testUser.setCreditCardListArray(new CreditCardList());
+        testUser.getCreditCardListArray().add(testCreditCard);
+        String input = "/cc 1 /n DBS /cb 5 /cl 2000 /bal 2000";
+        UpdateCommand updateCommand = new UpdateCommand(input, testUser);
+        try {
+            updateCommand.executeCommand();
+            assertEquals(testUser.getCreditCardListArray().get(0).toString(), newCreditCard.toString());
```

Comment:
Remove magic literal, remember to import the constant
```suggestion
            assertEquals(testUser.getCreditCardListArray().get(INDEX_OF_FIRST_ITEM).toString(), newCreditCard.toString());
```